src/ddmd/expression.d

```diff
@@ -11007,10 +11007,7 @@ extern (C++) final class NegExp : UnaExp
         if (tb.ty == Tarray || tb.ty == Tsarray)
         {
             if (!isArrayOpValid(e1))
-            {
-                error("invalid array operation %s (possible missing [])", toChars());
-                return new ErrorExp();
-            }
+                return arrayOpInvalidError(this);
```

Comment:
This maintains the simple true/false result of `isArrayOpValid()` and the purity of it making it simpler to understand.